chalice/awsclient.py

```diff
@@ -92,8 +119,12 @@ def create_function(self,
             kwargs['Timeout'] = timeout
         if memory_size is not None:
             kwargs['MemorySize'] = memory_size
-        return self._call_client_method_with_retries(
-            self._client('lambda').create_function, kwargs)['FunctionArn']
+        try:
+            return self._call_client_method_with_retries(
+                self._client('lambda').create_function, kwargs)['FunctionArn']
+        except Exception as e:
```

Comment:
Offline: Decided to just catch botocore.exceptions.ClientError and RequestsConnectionError for this.